pkg/features/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,97 @@
+package pagerduty
+
+import (
+	"fmt"
+	"os"
+	"slices"
+
+	"github.com/adrg/xdg"
+	"github.com/openshift/ocm-container/pkg/engine"
+	"github.com/openshift/ocm-container/pkg/features"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const (
+	defaultPagerDutyTokenFile = ".config/pagerduty-cli/config.json"
+	pagerDutyTokenDest        = "/root/" + defaultPagerDutyTokenFile
+)
+
+type Config struct {
+	Token string `mapstructure:"token_file"`
+	Mount string `mapstructure:"mount"`
+}
+
+func newConfigWithDefaults() *Config {
+	config := Config{}
+	config.Token = defaultPagerDutyTokenFile
+	config.Mount = "rw"
+
+	return &config
+}
+
+func (cfg *Config) validate() error {
+	validMountOptions := []string{
+		"ro",
+		"rw",
+	}
+	if !slices.Contains(validMountOptions, cfg.Mount) {
+		return fmt.Errorf("invalid mount option. Valid options are %s", validMountOptions)
+	}
+	return nil
+}
+
+func New() (features.OptionSet, error) {
```

Comment:
Idea here is that each feature independently gets its own config (not passed from the main func) and returns a well defined OptionSet.

Next steps - each feature handles its own "am I enabled" check and its own registration with the features controller.